pkg/distributors/contracts/DistributionScheduler.sol

```diff
@@ -17,104 +17,119 @@ pragma experimental ABIEncoderV2;
 
-import "./interfaces/IMultiDistributor.sol";
+import "./interfaces/IDistributionScheduler.sol";
 
 // solhint-disable not-rely-on-time
 
 /**
  * Scheduler for MultiDistributor contract
  */
-contract DistributionScheduler {
+contract DistributionScheduler is IDistributionScheduler {
     using SafeERC20 for IERC20;
 
     IMultiDistributor private immutable _multiDistributor;
+    mapping(bytes32 => ScheduledDistribution) private _scheduledDistributions;
 
     constructor(IMultiDistributor multiDistributor) {
         _multiDistributor = multiDistributor;
     }
 
-    enum DistributionStatus { UNINITIALIZED, PENDING, STARTED }
-
-    struct ScheduledDistribution {
-        bytes32 distributionId;
-        IERC20 stakingToken;
-        IERC20 distributionToken;
-        uint256 startTime;
-        address owner;
-        uint256 amount;
-        DistributionStatus status;
-    }
-
-    event DistributionScheduled(bytes32 indexed distributionId, bytes32 scheduleId, uint256 startTime, uint256 amount);
-    event DistributionStarted(bytes32 indexed distributionId, bytes32 scheduleId, uint256 startTime, uint256 amount);
-
-    mapping(bytes32 => ScheduledDistribution) private _scheduledDistributions;
-
-    function getScheduledDistributionInfo(bytes32 scheduleId) external view returns (ScheduledDistribution memory) {
+    function getScheduledDistributionInfo(bytes32 scheduleId)
+        external
+        view
+        override
+        returns (ScheduledDistribution memory)
+    {
         return _scheduledDistributions[scheduleId];
     }
 
-    function getScheduleId(
-        IERC20 stakingToken,
-        IERC20 distributionToken,
-        address owner,
-        uint256 startTime
-    ) public pure returns (bytes32) {
-        return keccak256(abi.encodePacked(stakingToken, distributionToken, owner, startTime));
+    function getScheduleId(bytes32 distributionId, uint256 startTime) public pure override returns (bytes32) {
+        return keccak256(abi.encodePacked(distributionId, startTime));
     }
 
     function scheduleDistribution(
         bytes32 distributionId,
-        IERC20 stakingToken,
-        IERC20 distributionToken,
         uint256 amount,
         uint256 startTime
-    ) public returns (bytes32 scheduleId) {
-        scheduleId = getScheduleId(stakingToken, distributionToken, msg.sender, startTime);
+    ) external override returns (bytes32 scheduleId) {
+        scheduleId = getScheduleId(distributionId, startTime);
         require(startTime > block.timestamp, "Distribution can only be scheduled for the future");
 
         require(
             _scheduledDistributions[scheduleId].status == DistributionStatus.UNINITIALIZED,
             "Distribution has already been scheduled"
         );
 
+        // As funding pushes out the end timestamp of the distribution channel
+        // we only allow the distribution owner to schedule distributions
```

Comment:
I think the problem is not so much that the duration is reset (which should be fine), but rather that tokens pending distribution are rolled-over into the new, extended duration (e.g. if there's one day pending worth of rewards, re-starting the distribution with 1 wei would dillute that day worth of tokens over e.g. the full week).